From 9e4a2f03bb37ec34f9e50c69b792974531056780 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Wed, 2 Dec 2020 07:44:32 +0200
Subject: [PATCH] cm_fx6: env: Update the SOM boot environmen

1) Add ethactusb environment command.
2) Change the preboot logics.
Issue usb start if ext_phy only.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 include/configs/cm_fx6.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/configs/cm_fx6.h b/include/configs/cm_fx6.h
index a1e79cee34..55ff1da8c7 100644
--- a/include/configs/cm_fx6.h
+++ b/include/configs/cm_fx6.h
@@ -142,7 +142,8 @@
 		"run setboottypem;" \
 		"run trybootk;" \
 		"run setboottypez;" \
-		"run trybootk;\0"
+		"run trybootk;\0" \
+	"ethactusb=test ! -z \"${ext_phy}\" && test ! -z \"${usb_net}\" && setenv ethact ${usb_net};\0"
 
 #ifdef CL_SOM_IMX6
 #define EMMC "setenv storagedev 3;"\
@@ -172,7 +173,7 @@
 	"run setupnandboot;" \
 	"run nandboot;"
 
-#define CONFIG_PREBOOT		"usb start"
+#define CONFIG_PREBOOT		"usb start; run ethactusb"
 
 /* SPI */
 #define CONFIG_SPI
-- 
2.11.0

